lib/options_validator.js

```diff
@@ -46,48 +52,57 @@ function validationFunction(optionsSchema, providedOptions, validationOptions) {
 
   const verifiedOptions = Object.assign({}, providedOptions);
 
-  if (validationLevel === VALIDATION_LEVEL_NONE) {
-    return Object.freeze(verifiedOptions);
-  }
-
   Object.keys(optionsSchema).forEach(optionName => {
     const optionFromSchema = optionsSchema[optionName];
 
-    if (optionFromSchema.required && !verifiedOptions.hasOwnProperty(optionName)) {
-      const invalidateMessageRequired = `required option [${optionName}] was not found.`;
-      invalidate(validationLevel, invalidateMessageRequired, logger);
+    if (overrideOptions && overrideOptions[optionName] != null) {
+      if (!verifiedOptions.hasOwnProperty(optionName)) {
+        verifiedOptions[optionName] = overrideOptions[optionName];
+      }
+      if (optionFromSchema && optionFromSchema.default) {
+        console.warn(
```

Comment:
I could see this warning confusing users. Something like:

```js
const coll = client.db('foo', { readPreference: ReadPreference.secondary }).collection('bar');
const result = await coll.findOne({...}); // might console.warn if there is a "default" for readPreference
```

Should we be warning here?

Reply:
In this case, the "default" for `readPreference` would actually be handled by `overrideOptions` so we would have no `default` value for `readPreference`, but we would have `overrideOptions = { readPreference: db.readPreference }`. There shouldn't be anywhere that we have both a `default` and an `overrideOption`.